src/inference/mh.jl

```diff
@@ -62,33 +62,34 @@ end
 
-The `involution` Julia function has the following signature:
+The `involution` is a callable value that has the following signature:
 
-    (new_trace, bwd_choices::ChoiceMap, weight) = involution(trace, fwd_choices::ChoiceMap, fwd_ret, proposal_args::Tuple)
+    (new_trace, bwd_choices::ChoiceMap, weight) = involution(trace, fwd_choices::ChoiceMap, fwd_ret, proposal_args::Tuple; check::Bool=false)
+
+Most users will want to construct `involution` using the [Involution DSL](@ref) with the [`@involution`](@ref) macro, but is also possible to provide a Julia function for `involution`.
 
 The generative function `proposal` is executed on arguments `(trace, proposal_args...)`, producing a choice map `fwd_choices` and return value `fwd_ret`.
 For each value of model arguments (contained in `trace`) and `proposal_args`, the `involution` function applies an involution that maps the tuple `(get_choices(trace), fwd_choices)` to the tuple `(get_choices(new_trace), bwd_choices)`.
 Note that `fwd_ret` is a deterministic function of `fwd_choices` and `proposal_args`.
 When only discrete random choices are used, the `weight` must be equal to `get_score(new_trace) - get_score(trace)`.
-
-**Including Continuous Random Choices**
-When continuous random choices are used, the `weight` must include an additive term that is the determinant of the the Jacobian of the bijection on the continuous random choices that is obtained by currying the involution on the discrete random choices.
+When continuous random choices are used, the `weight` returned by the involution must include an additive term that is the determinant of the the Jacobian of the bijection on the continuous random choices that is obtained by currying the involution on the discrete random choices.
+The `check` keyword argument to the involution can be used to enable or disable any dynamic correctness checks that the involution performs; for successful executions, `check` does not alter the return value.
 """
 function metropolis_hastings(
         trace, proposal::GenerativeFunction,
-        proposal_args::Tuple, involution::Function;
+        proposal_args::Tuple, involution;
         check=false, observations=EmptyChoiceMap())
     (fwd_choices, fwd_score, fwd_ret) = propose(proposal, (trace, proposal_args...,))
     (new_trace, bwd_choices, weight) = involution(trace, fwd_choices, fwd_ret, proposal_args)
     (bwd_score, bwd_ret) = assess(proposal, (new_trace, proposal_args...), bwd_choices)
     check && check_observations(get_choices(new_trace), observations)
     if check
-        (trace_rt, fwd_choices_rt, weight_rt) = involution(new_trace, bwd_choices, bwd_ret, proposal_args)
+        (trace_rt, fwd_choices_rt, weight_rt) = involution(new_trace, bwd_choices, bwd_ret, proposal_args; check=check)
         if !isapprox(fwd_choices_rt, fwd_choices)
             println("fwd_choices:")
             println(fwd_choices)
```

Comment:
I'd like to start moving from `println` to use of the logging macros, e.g. here:
```suggestion
            @error "fwd_choices: $fwd_choices"
```
and similarly below (but I'm not allowed to suggest below since it's not close to the changed lines).

Reply:
@bzinberg This code was already there. Let's create a separate issue for this.

Reply:
No need -- this change was included in https://github.com/probcomp/Gen/commit/5d3938572fe5f0e9ab468729a394c237695f2bfd and I don't think there are any more instances of it.